static/js/components/dialogs/CollectionFormDialog.js

```diff
@@ -76,28 +78,67 @@ class CollectionFormDialog extends React.Component {
     dispatch(uiActions.setAdminLists(event.target.value));
   };
 
+  validate = () => {
+    const { collectionForm } = this.props;
+    const validation = validateCollectionForm(this.getPayload(collectionForm));
+    this.handleError(validation);
+  };
+
+  getPayload = (collectionForm: CollectionFormState) => {
```

Comment:
Can you set a return type for this function?

Reply:
Decided to revert this to no longer be a function.  At one point during development of this PR I had been calculating the payload in more than one place but that's no longer true.